File: frontend/packages/shared-api/src/admin/collection.ts
import type { AxiosInstance } from 'axios';
import type { ApiResponse, PaginatedResponse } from '@shared/types';

export interface LixiaoyunRawCompanyRow {
  id: string;
  pid: string | null;
  keyword_master_id: string | null;
  keyword: string | null;
  entname: string | null;
  entname_eng: string | null;
  esdate: string | null;
  reg_cap: string | null;
  official_website: string | null;
  regccap: string | null;
  scale: string | null;
  annual_turnover: string | null;
  legalperson: string | null;
  geo_address: string | null;
  dom: string | null;
  collected_at: string | null;
}

export interface LixiaoyunApiCompanyDetail extends LixiaoyunRawCompanyRow {
  provider?: string;
  entstatus?: string | null;
  enttype?: string | null;
  opscope?: string | null;
  industryphy_desc?: string | null;
  secindustry_desc?: string[] | string | null;
  industry_l3_desc?: string | null;
  industry_l4_desc?: string | null;
  uncid?: string | null;
  ent_introduction?: string | null;
  opfrom?: string | null;
  opto?: string | null;
  regorg?: string | null;
  apprdate?: string | null;
  oploc?: string | null;
}

export interface WaimaotongRawCompanyRow {
  id: string;
  company_name: string | null;
  country: string | null;
  domain: string | null;
  industry: string | null;
  employee_size: string | null;
  founded_year: number | null;
  full_address: string | null;
  source_keyword: string | null;
  source_competitor: string | null;
  source_type: string | null;
  contacts_count: number | null;
  email_count: number | null;
  has_detail: boolean | null;
  has_contacts: boolean | null;
  id_verified: boolean | null;
  website: string | null;
  api_company_id: string | null;
  created_at: string;
}

export interface WaimaotongRawContactRow {
  id: string;
  raw_company_id: string;
  source_contact_id: string | null;
  name: string | null;
  position: string | null;
  department: string | null;
  email: string | null;
  email_status: string | null;
  phone: string | null;
  linkedin: string | null;
  source: string | null;
  confidence: number | null;
  created_at: string;
}

export interface WmtCleanCompanyRow {
  id: string;
  source_id: string | null;
  name: string | null;
  company_name: string | null;
  english_name: string | null;
  country: string | null;
  country_iso3: string | null;
  domain: string | null;
  industry: string | null;
  sub_industry: string | null;
  phone: string | null;
  employee_size: string | null;
  company_size: string | null;
  founded_year: number | null;
  website: string | null;
  full_address: string | null;
  description: string | null;
  grade: string | null;
  score: number | null;
  email_priority: string | null;
  company_type_analysis: string | null;
  product_tags: string[];
  data_source_tags: string[];
  collection_type: 'manual' | 'keyword' | 'reverse' | 'unknown';
  has_trade_data: boolean | null;
  trade_amount_3y_usd: number | null;
  trade_count: number | null;
  contacts_count: number | null;
  detail_status: string | null;
  contacts_status: string | null;
  trade_status: string | null;
  sys_company_id: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface WmtCleanCompanyDetail extends WmtCleanCompanyRow {
  score_details: unknown[] | null;
  match_reasons: unknown[] | null;
  potential_needs: unknown[] | null;
  recommended_products: unknown[] | null;
  risk_factors: unknown[] | null;
  main_business: unknown[] | null;
  trade_summary: Record<string, unknown> | null;
  sales_approach: string | null;
}

export interface WmtCleanContactRow {
  id: string;
  name: string | null;
  position: string | null;
  department: string | null;
  email: string | null;
  email_status: string | null;
  phone: string | null;
  mobile: string | null;
  linkedin: string | null;
  whatsapp: string | null;
  source: string | null;
  confidence: number | null;
  created_at: string | null;
}

export interface LixiaoyunCleanCompanyKeyword {
  keyword_master_id: string;
  keyword: string;
  keyword_normalized: string;
}

export interface LixiaoyunCleanCompanyRow {
  id: string;
  pid: string | null;
  entname: string | null;
  entname_eng: string | null;
  esdate: string | null;
  reg_cap: string | null;
  official_website: string | null;
  regccap: string | null;
  scale: string | null;
  annual_turnover: string | null;
  legalperson: string | null;
  geo_address: string | null;
  dom: string | null;
  industry_tag: string | null;
  keyword_master: LixiaoyunCleanCompanyKeyword[];
  created_at: string | null;
}

export interface LixiaoyunCleanCompanyDetail extends LixiaoyunCleanCompanyRow {
  uncid: string | null;
  enttype: string | null;
  enttype_code: string | null;
  entstatus: string | null;
  entstatus_code: number | null;
  regno: string | null;
  organizational_code: string | null;
  opfrom: number | null;
  opto: number | null;
  regorg: string | null;
  apprdate: number | null;
  revokedate: number | null;
  province: number | null;
  city: number | null;
  district: number | null;
  reg_province: number | null;
  reg_city: number | null;
  reg_district: number | null;
  oploc: string | null;
  industryphy: string | null;
  industryphy_desc: string | null;
  opscope: string | null;
  secindustry: unknown;
  secindustry_desc: unknown;
  industry_l3: string | null;
  industry_l3_desc: string | null;
  industry_l4: string | null;
  industry_l4_desc: string | null;
  historyname_list: unknown;
  legalperson_desc: string | null;
  location_code: string | null;
  updated_at: string | null;
}

export function collectionApi(client: AxiosInstance) {
  return {
    listLixiaoyunRawCompanies: (params: {
      page?: number;
      page_size?: number;
      keyword?: string;
      keyword_filter?: string;
      found_date_start?: string;
      found_date_end?: string;
      reg_capital?: string;
      employee_scale?: string;
      has_name_en?: boolean;
      has_domain?: boolean;
    }) =>
      client.get<PaginatedResponse<LixiaoyunRawCompanyRow>>(
        '/api/v1/raw/lixiaoyun/companies',
        { params },
      ),
    getLixiaoyunRawCompanyDebug: (rawCompanyId: string) =>
      client.get<ApiResponse<LixiaoyunApiCompanyDetail>>(
        `/api/v1/raw/lixiaoyun/companies/${encodeURIComponent(rawCompanyId)}/debug`,
      ),
    listWaimaotongRawCompanies: (params: {
      page?: number;
      page_size?: number;
      q?: string;
      country?: string;
      source_keyword?: string;
      source_competitor?: string;
      industry?: string;
      size?: string;
      year_min?: number;
      year_max?: number;
      has_contacts?: boolean;
    }) =>
      client.get<PaginatedResponse<WaimaotongRawCompanyRow>>(
        '/api/v1/raw/waimaotong/companies',
        { params },
      ),
    getWaimaotongRawCompanyDebug: (rawCompanyId: string) =>
      client.get<ApiResponse<Record<string, unknown>>>(
        `/api/v1/raw/waimaotong/companies/${encodeURIComponent(rawCompanyId)}/debug`,
      ),
    listWaimaotongRawCompanyContacts: (rawCompanyId: string) =>
      client.get<PaginatedResponse<WaimaotongRawContactRow>>(
        `/api/v1/raw/waimaotong/companies/${encodeURIComponent(rawCompanyId)}/contacts`,
      ),
    listLixiaoyunCleanCompanies: (params: {
      page?: number;
      page_size?: number;
      keyword?: string;
      keyword_filter?: string;
      industry_tag?: string;
      found_date_start?: string;
      found_date_end?: string;
      reg_capital?: string;
      employee_scale?: string;
      has_name_en?: boolean;
      has_domain?: boolean;
    }) =>
      client.get<PaginatedResponse<LixiaoyunCleanCompanyRow>>(
        '/api/v1/collection/lixiaoyun-clean-companies',
        { params },
      ),
    getLixiaoyunCleanCompanyDetail: (id: number) =>
      client.get<ApiResponse<LixiaoyunCleanCompanyDetail>>(
        `/api/v1/collection/lixiaoyun-clean-companies/${id}`,
      ),
    listWmtCleanCompanies: (params: {
      page?: number;
      page_size?: number;
      q?: string;
      country?: string;
      industry?: string;
      size?: string;
      year_min?: number;
      year_max?: number;
      has_contacts?: boolean;
      grade?: string;
      collection_type?: string;
    }) =>
      client.get<PaginatedResponse<WmtCleanCompanyRow>>(
        '/api/v1/collection/wmt-clean-companies',
        { params },
      ),
    getWmtCleanCompany: (companyId: number) =>
      client.get<ApiResponse<WmtCleanCompanyDetail>>(
        `/api/v1/collection/wmt-clean-companies/${companyId}`,
      ),
    listWmtCleanCompanyContacts: (companyId: number) =>
      client.get<PaginatedResponse<WmtCleanContactRow>>(
        `/api/v1/collection/wmt-clean-companies/${companyId}/contacts`,
      ),
  };
}

File: backend/app/api/admin/collection.py
from fastapi import APIRouter, Depends, Query

from app.core.responses import paginated_response, success_response
from app.security.dependencies import PlatformAuthContext, get_current_platform_user
from app.services.admin_collection_service import AdminCollectionService

router = APIRouter(tags=["admin-collection"])
service = AdminCollectionService()


@router.get("/raw/{provider}/companies")
async def list_v3_raw_companies(
    provider: str,
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=1000),
    keyword_master_id: str | None = None,
    q: str | None = None,
    keyword: str | None = None,
    country_iso3: str | None = None,
    country: str | None = None,
    # lixiaoyun
    keyword_filter: str | None = None,
    found_date_start: str | None = None,
    found_date_end: str | None = None,
    reg_capital: str | None = None,
    employee_scale: str | None = None,
    contacts_filter: str | None = None,
    has_name_en: bool | None = None,
    has_domain: bool | None = None,
    # tendata
    industry: str | None = None,
    tag: str | None = None,
    size: str | None = None,
    amount_min: float | None = None,
    amount_max: float | None = None,
    count_min: int | None = None,
    count_max: int | None = None,
    pcb: str | None = None,
    contact_min: int | None = None,
    contact_max: int | None = None,
    year_min: int | None = None,
    year_max: int | None = None,
    method: str | None = None,
    # waimaotong
    source_keyword: str | None = None,
    source_competitor: str | None = None,
    has_contacts: bool | None = None,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    rows, total = await service.list_v3_raw_companies(
        context.connection,
        provider=provider,
        page=page,
        page_size=page_size,
        keyword_master_id=keyword_master_id,
        q=q or keyword,
        country_iso3=country_iso3 or country,
        keyword_filter=keyword_filter,
        found_date_start=found_date_start,
        found_date_end=found_date_end,
        reg_capital=reg_capital,
        employee_scale=employee_scale,
        contacts_filter=contacts_filter,
        has_name_en=has_name_en,
        has_domain=has_domain,
        industry=industry,
        tag=tag,
        size=size,
        amount_min=amount_min,
        amount_max=amount_max,
        count_min=count_min,
        count_max=count_max,
        pcb=pcb,
        contact_min=contact_min,
        contact_max=contact_max,
        year_min=year_min,
        year_max=year_max,
        method=method,
        source_keyword=source_keyword,
        source_competitor=source_competitor,
        has_contacts=has_contacts,
    )
    return paginated_response(rows, total=total, has_more=(page * page_size < total))


@router.get("/raw/{provider}/companies/{raw_company_id}/debug")
async def get_v3_raw_company_debug(
    provider: str,
    raw_company_id: int,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    result = await service.get_v3_raw_company_debug(
        context.connection,
        provider=provider,
        raw_company_id=raw_company_id,
    )
    return success_response(result)


@router.get("/raw/{provider}/companies/{raw_company_id}/contacts")
async def list_v3_raw_company_contacts(
    provider: str,
    raw_company_id: int,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    rows = await service.list_v3_raw_company_contacts(
        context.connection,
        provider=provider,
        raw_company_id=raw_company_id,
    )
    return paginated_response(rows, total=len(rows))


# ── 同行公司（清洗）──────────────────────


@router.get("/collection/lixiaoyun-clean-companies")
async def list_lixiaoyun_clean_companies(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=1000),
    keyword: str | None = None,
    keyword_filter: str | None = None,
    industry_tag: str | None = None,
    found_date_start: str | None = None,
    found_date_end: str | None = None,
    reg_capital: str | None = None,
    employee_scale: str | None = None,
    has_name_en: bool | None = None,
    has_domain: bool | None = None,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    rows, total = await service.list_lixiaoyun_clean_companies(
        context.connection,
        page=page,
        page_size=page_size,
        keyword=keyword,
        keyword_filter=keyword_filter,
        industry_tag=industry_tag,
        found_date_start=found_date_start,
        found_date_end=found_date_end,
        reg_capital=reg_capital,
        employee_scale=employee_scale,
        has_name_en=has_name_en,
        has_domain=has_domain,
    )
    return paginated_response(rows, total=total, has_more=(page * page_size < total))


@router.get("/collection/lixiaoyun-clean-companies/{company_id}")
async def get_lixiaoyun_clean_company_detail(
    company_id: int,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    result = await service.get_lixiaoyun_clean_company_detail(
        context.connection, company_id=company_id
    )
    return success_response(result)


# ── waimaotong clean companies ──────────────────────────────────


@router.get("/collection/wmt-clean-companies")
async def list_wmt_clean_companies(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=1000),
    q: str | None = None,
    country: str | None = None,
    industry: str | None = None,
    size: str | None = None,
    year_min: int | None = None,
    year_max: int | None = None,
    has_contacts: bool | None = None,
    grade: str | None = None,
    collection_type: str | None = None,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    rows, total = await service.list_wmt_clean_companies(
        context.connection,
        page=page,
        page_size=page_size,
        q=q,
        country=country,
        industry=industry,
        size=size,
        year_min=year_min,
        year_max=year_max,
        has_contacts=has_contacts,
        grade=grade,
        collection_type=collection_type,
    )
    return paginated_response(rows, total=total, has_more=(page * page_size < total))


@router.get("/collection/wmt-clean-companies/{company_id}")
async def get_wmt_clean_company(
    company_id: int,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    result = await service.get_wmt_clean_company(
        context.connection,
        company_id=company_id,
    )
    return success_response(result)


@router.get("/collection/wmt-clean-companies/{company_id}/contacts")
async def list_wmt_clean_company_contacts(
    company_id: int,
    context: PlatformAuthContext = Depends(get_current_platform_user),
) -> dict:
    rows = await service.list_wmt_clean_company_contacts(
        context.connection,
        company_id=company_id,
    )
    return paginated_response(rows, total=len(rows))
